compiler/luci/service/include/luci/Service/CircleShapeSignatureInferenceHelper.h

```diff
@@ -26,8 +26,16 @@ namespace luci
 namespace ssinf // Namespace for Shape Signature Inference
 {
 
+// Return empty signature if all of dimensions are known.
+// If one of dimensions is unknown, return signature without change.
+ShapeSignature clean_signature(const luci::ShapeSignature &signature);
```

Comment:
I first though of this to clean(? not sure what the meaning of 'clean' maybe) the signature values... but actually it isn't.

Reply:
I think we need to use another name than `clean`... let's take some time and think about this...

Reply:
`normalize_` `validate_` regularize_` `legalize_`
I've seen `legalize` in LLVM that makes things make sense. how about `legalize_signature` ?

Reply:
Always thanks for good idea :-)